nemo/collections/asr/models/clustering_diarizer.py

```diff
@@ -316,6 +324,10 @@ def _perform_speech_activity_detection(self):
             raise ValueError(
                 "Only one of diarizer.oracle_vad, vad.model_path or vad.external_vad_manifest must be passed"
             )
+        if self.AUDIO_RTTM_MAP == {}:
+            raise ValueError(
+                "self.AUDIO_RTTM_MAP is empty: There are no input audio files containing detected speech signal. Abort."
```

Comment:
No. It can't be moved into `_run_vad()` without repetition. Empty audio can be happening in either `_run_vad()` or `write_rttm2manifest()` function.  This line ` if self.AUDIO_RTTM_MAP == {}` should be after VAD case handling  to avoid repetition.